src/component/Text.tsx

```diff
@@ -183,109 +183,113 @@ const getWordsByLines = (props: Props, needCalculate: boolean) => {
   return getWordsWithoutCalculate(props.children);
 };
 
-export class Text extends Component<Props, State> {
-  static defaultProps = {
-    x: 0,
-    y: 0,
-    lineHeight: '1em',
-    capHeight: '0.71em', // Magic number from d3
-    scaleToFit: false,
-    textAnchor: 'start',
-    verticalAnchor: 'end', // Maintain compat with existing charts / default SVG behavior
-    fill: '#808080',
-  };
+const textDefaultProps = {
+  x: 0,
+  y: 0,
+  lineHeight: '1em',
+  capHeight: '0.71em', // Magic number from d3
+  scaleToFit: false,
+  textAnchor: 'start',
+  verticalAnchor: 'end', // Maintain compat with existing charts / default SVG behavior
+  fill: '#808080',
+};
 
-  state: State = {};
+export const Text = (props: Props) => {
+  const [state, setState] = useState<State>({
+    prevWidth: props.width,
+    prevScaleToFit: props.scaleToFit,
+    prevChildren: props.children,
+    prevStyle: props.style,
+    wordsByLines: getWordsByLines(props, true),
```

Comment:
Instead of building on state object, you could simply have separate state variables.

Reply:
It would also be nice to avoid recreating the initial state
https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state

Reply:
Now I think we don't need the state at all, its just a trigger for recalculation, I tried to keep the code as is mostly, but now I think a useMemo would be sufficient, it calculates on first render, and when the props changed it'll recalculate the wordsByLine. not need to state+effect.

Reply:
💪